src/lib.rs

```diff
@@ -61,69 +76,253 @@ impl InlinerBuilder {
+    pub fn is_initial(&self) -> bool {
+        match self {
+            Error::Root(_) => true,
+            Error::Inline(_) => false,
+        }
+    }
+
+    /// Returns true if the error happened while inlining modules.
+    pub fn is_inline(&self) -> bool {
+        match self {
+            Error::Root(_) => false,
+            Error::Inline(_) => true,
+        }
+    }
+}
+
+/// The kind of error that was encountered for a particular file.
 #[derive(Debug)]
-pub enum Error {
-    /// The contents for one or more modules could not be found
-    NotFound(Vec<(String, SourceLocation)>),
-    #[doc(hidden)]
-    __NonExhaustive,
+pub enum ErrorKind {
+    /// An error happened while opening or reading the file.
+    Io(io::Error),
+
+    /// Errors happened while using `syn` to parse the file.
+    Parse(syn::Error),
+}
+
+impl From<io::Error> for ErrorKind {
+    fn from(err: io::Error) -> Self {
+        ErrorKind::Io(err)
+    }
+}
+
+impl From<syn::Error> for ErrorKind {
+    fn from(err: syn::Error) -> Self {
+        ErrorKind::Parse(err)
+    }
+}
+
+impl fmt::Display for ErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ErrorKind::Io(err) => write!(f, "IO error: {}", err),
+            ErrorKind::Parse(err) => write!(f, "parse error: {}", err),
+        }
+    }
+}
+
+/// An inlining that was partially - but not completely - successful. The root file was parsed
+/// successfully, but 1 or more referenced modules failed to load.
+pub struct IncompleteInlining {
+    output: syn::File,
+    errors: Vec<InlineError>,
+}
+
+impl IncompleteInlining {
+    /// Create a new `IncompleteInlining` with the best-effort output and the list of errors
+    /// encountered.
+    ///
+    /// # Panics
+    /// This function will panic if `errors` is empty. An empty error list means the inlining
+    /// is complete, and this struct should not be used.
+    pub(crate) fn new(output: syn::File, errors: Vec<InlineError>) -> Self {
+        if errors.is_empty() {
+            panic!("Inlining must encounter 1 or more errors to be incomplete");
+        }
+
+        Self { output, errors }
+    }
+
+    /// The best-effort result of inlining.
+    pub fn output(&self) -> &syn::File {
+        &self.output
+    }
+
+    /// The errors that kept the inlining from completing. This is guaranteed not to be empty.
+    pub fn errors(&self) -> &[InlineError] {
+        &self.errors
+    }
+
+    /// Break an incomplete inlining into the best-effort parsed result and the errors encountered.
+    ///
+    /// # Usage
+    ///
+    /// ```rust,norun
+    /// # use syn_inline_mod::{Error, IncompleteInlining};
+    /// # #[allow(unused_vars)]
+    /// # fn handle_error(error: Error) {
+    /// if let Error::Inline(err) = error {
+    ///     let (output, errors) = err.into_output_and_errors();
+    /// }
+    /// # }
+    /// ```
```

Comment:
This really suggests that maybe it can just be a struct variant on the enum directly I think.

Reply:
That's a reasonable point in favor of no struct. Let's consolidate this discussion on the other thread about `IncompleteInlining` though.